maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -1232,11 +1239,6 @@ private void deepCopy( MavenProject project )
             setDependencyArtifacts( Collections.unmodifiableSet( project.getDependencyArtifacts() ) );
         }
 
-        if ( project.getArtifacts() != null )
-        {
-            setArtifacts( Collections.unmodifiableSet( project.getArtifacts() ) );
```

Comment:
This didn't make any sense anymore because the entire holder is now copied already (see further up).